### PR TITLE
Dialogs: add offline repo HTML form 

### PR DESCRIPTION
**Pull request by mbtools**
"Add online" is working well. Time to convert the offline version to HTML form as well:

image

### CODE REVIEW

**Comment by larshp** [comment]
see https://github.com/larshp/abapGit/issues/3417 regarding the "add online" page not working well

**Review by larshp — DISMISSED** [review]
await https://github.com/larshp/abapGit/pull/3848

**Comment by larshp** [comment]
the old `zcl_abapgit_ui_factory=>get_popups( )->repo_new_offline( )` can be deleted?

File: src/ui/zcl_abapgit_services_repo.clas.abap [hunk 1]
```diff
@@ -206,12 +198,15 @@ CLASS zcl_abapgit_services_repo IMPLEMENTATION.
       iv_ign_subpkg       = is_repo_params-ignore_subpackages
       iv_master_lang_only = is_repo_params-master_lang_only ).
 
+    " Make sure there're no leftovers from previous repos
+    ro_repo->rebuild_local_checksums( ).
```
Comment (sbcgua) [inline src/ui/zcl_abapgit_services_repo.clas.abap:202]: Mmm, a new repo is empty ... is it needed ?

**Review by sbcgua — APPROVED** [review]
Seems to work well.

One comment above, not critical just to double check.

P.S. I'm usually against subclassing but in this case it may be a beneficial next refactoring step. A lot of code duplication with online. But pls wait with it, I have one html_form refactoring idea

File: src/ui/zcl_abapgit_services_repo.clas.abap (hunk 1 above)
Comment (mbtools) [reply src/ui/zcl_abapgit_services_repo.clas.abap:202]: The rebuild is for local objects which will exist if the package for the repo already contains objects. It was there before, so I kept it.

Comment (mbtools) [reply src/ui/zcl_abapgit_services_repo.clas.abap]: It rebuilds for local objects, which could exist already. But there's most likely a serialize step triggered afterward that does the checksums anyway

**Comment by larshp** [comment]
@mbtools check conflicts

**Comment by sbcgua** [comment]
> P.S. ... But pls wait with it, I have one html_form refactoring idea

have a look at #3904

P.P.S sorry for messing the base code :(

**Comment by mbtools** [comment]
No worry. I will wait

**Comment by sbcgua** [comment]
@mbtools will you wait for completion of #3602 ? (postdata in particular)
Imho this is good and useful code to merge. #3602 can take some more time. I'd just suggest taking into account ideas from #3904 (but it's more or less moving the definition of the form to constructor)

**Comment by mbtools** [comment]
I will adjust this one once your phase 3 is merged

**Comment by mbtools** [comment]
Closing to do it properly

**Pull request closed without merging**